tests/components/CopyPaste/CopyPaste.spec.ts

```diff
@@ -71,17 +87,59 @@ describe('CopyToClipboard ResultAction', () => {
         });
+
+        describe('when the clipboard api is not available', () => {
+            beforeEach(() => {
+                writeTextStub.get(() => undefined);
+            });
+
+            it('should change the tooltip text for "Copy"', () => {
+                testComponent.element.querySelector<HTMLElement>('.coveo-caption-for-icon').innerText = 'patate';
+                testComponent.element.dispatchEvent(new MouseEvent('mouseleave'));
+                expect(testComponent.element.querySelector<HTMLElement>('.coveo-caption-for-icon').innerText).toEqual(l('CopyToClipboard_copy'));
+            });
+        });
     });
 
-    describe('the template option', () => {
-        it('should use the provided template to fill the clipboard', () => {
-            testComponent.options.template = 'foo bar ${title}';
-            testComponent.element.click();
-            expect(writeTextStub.args[0][0]).toEqual(`foo bar ${testComponent.result.title}`);
+    describe('options', () => {
+        describe('template', () => {
+            it('should use the provided template to fill the clipboard', () => {
+                buildResultAction({ template: 'foo bar ${title}' });
+                testComponent.element.click();
+                expect(writeTextStub.args[0][0]).toEqual(`foo bar ${testComponent.result.title}`);
+            });
+
+            it('should use the "{title}\\n{clickUri}" template by default', () => {
+                testComponent.element.click();
+                expect(writeTextStub.args[0][0]).toEqual(`${testComponent.result.title}\n${testComponent.result.clickUri}`);
+            });
         });
-        it('should use the "{title}\\n{clickUri}" template by default', () => {
-            testComponent.element.click();
-            expect(writeTextStub.args[0][0]).toEqual(`${testComponent.result.title}\n${testComponent.result.clickUri}`);
+
+        describe('tooltip', () => {
+            it('should use the provided tooltip', () => {
+                const tooltipText = 'some tooltip';
+                buildResultAction({ tooltip: tooltipText });
+                expect(testComponent.element.querySelector<HTMLElement>('.coveo-caption-for-icon').innerText).toEqual(tooltipText);
+            });
+        });
+
+        describe('successTooltip', () => {
+            it('should use the provided successTooltip', async () => {
+                const tooltipText = 'some success tooltip';
+                buildResultAction({ successTooltip: tooltipText });
+                testComponent.element.click();
+
+                await Promise.resolve();
+                expect(testComponent.element.querySelector<HTMLElement>('.coveo-caption-for-icon').innerText).toEqual(tooltipText);
+            });
+
+            it('should not change the tooltip if the successTooltip is empty-like', async () => {
```

Comment:
Why not `...if the successTooltip is empty.`? Using `empty-like` hints me that you should have tested `undefined` as well.

Reply:
null undefined or '' is the same

Reply:
I know but your test is checking only one of those 3 cases right now. The best would be to validate those 3 cases. If we stick to validating only the empty string, I would rename the test case to make it clear.

Reply:
oki